Individual Treatment Effect

riccardocadei opened the pull request:
Significantly modify ITE estimators (both methods and function).

Added:
- S-Learner (`slearner`) method for ITE estimation  (fix Issue #96).
- T-Learner (`tlearner`) method for ITE estimation  (fix Issue #96).
- X-Learner (`xlearner`) method for ITE estimation  (fix Issue #96).

Rename:
- `offset` method-parameter -> hyper-parameter.
- `estimate_ite_poisson` function -> `estimate_ite_tpoisson`  (fix Issue #96).

Discard: 
- `include_ps_inf` method-parameter.
- `include_ps_dis` method-parameter.
- `oreg` method for ITE estimation.
- `ipw` method for ITE estimation (fix Issue #95).
- `sipw` method for ITE estimation (fix Issue #95).
- ITE standard deviation estimation.

Clean and discard many useless internal parameters for ITE estimation.

Naeemkh reviewed (changes requested):
Thanks, @riccardocadei :

- Please see my comments (mostly on styling).
- Please add a note section to the CRE package and direct the users to read more about S-, T-, and X-Learners.
- In the CRE parameters, for `replace`, please add more details. Replacing what and where in the process. 
- Do you have any example using `interventional_vars` in the CRE function? It does not show up in the default parameters. 
- CRE function documentation and default values (for cutoff and pfer) are not consistent. Please double-check.

riccardocadei commented on NEWS.md:
```diff
@@ -25,13 +32,23 @@ estimations.
 * `include_offset` method parameter.
 
+## Solved
+* Rules Generation Issue.
```
See Issue #91.

Naeemkh replied:
It should be clear to the users, too.

riccardocadei commented on R/filter_irrelevant_rules.R:
```diff
@@ -30,8 +28,13 @@ filter_irrelevant_rules <- function(rules, X, ite, max_decay, type_decay) {
+
+  for (i in 1:length(rules)) {
+    if (!grepl("&", rules[i]) & grepl("<=", rules[i])) {
+      rules[i] <- sub("<=", ">", rules[i])
```
yes, since we are considering only rules with lengths==1 .

riccardocadei commented on vignettes/CRE.Rmd:
```diff
@@ -47,26 +47,23 @@ __Data (required)__
 __Parameters (not required)__    
 **`method_parameters`** The list of parameters to define the models used, including:
```
between 47 and 48?

riccardocadei commented:
CRE function documentation and default values (for cutoff and pfer) seems consistent to me:
pfer = 1, cutoff = 0.9.
Am I missing something?

Naeemkh commented:
> 

In the default values, it is the other way around:
pfer = 0.9, cutoff = 1.

riccardocadei commented:
great catch on the default values!

I think I have addressed all your comments. Except leaving a reference for the *-learners. I think they are already well-known in the literature, and if not, we will leave the proper references in the main paper (and software paper) if necessary.
I don't see why adding a reference for them, not others.

Naeemkh commented:
> great catch on the default values!
> 
> I think I have addressed all your comments. Except leaving a reference for the *-learners. I think they are already well-known in the literature, and if not, we will leave the proper references in the main paper (and software paper) if necessary. I don't see why adding a reference for them, not others.

That is fine; we can skip the reference. To answer your question, we use their packages for other methods, but for these three approaches, we code them.
